### DISCUSSION TITLE
Synology NAS installation

### DISCUSSION THREAD

**Discussion by klausdieterf** [kimai/kimai#4203] [On-Premise: Installation & Updates] [answered]
I tried to install Kimai based on a documentation by Wolfgang, I stopped at page 4.
It looks like my installation is a newer version of Web Station and therefore the description no longer fits.
Can Wolfgang update the "Installationsanleitung" for the new Web Station.
**My Web Station**
![Webstation](https://github.com/kimai/kimai/assets/140513575/2e1b258d-c517-45b3-b879-caf36f7a6998)
**Web Station in the "Installationsanleitung"**
![Webstation docu](https://github.com/kimai/kimai/assets/140513575/b991e502-0e59-480c-8d00-90283963d8d4)
**Installed Web Station on my Synology**
![Web Station installed](https://github.com/kimai/kimai/assets/140513575/8d34b701-66b6-45c3-8da0-2d36319980bb)

**Comment by buzter** [comment]
You need PHP 8.1 or above for Kimai 2.0.xx. Your PHP version seems to be 7.4.

**Comment by SaturnusDJ** [comment]
I recommend Docker. Easy management. Have not found a downside yet for it in our organisation.

**Comment by BeckeBauer** [comment, marked as answer]
Nginx management has become even more annoying after the update to DSM 7:

Prerequisite:  Install the packages Web Station and PHP 8.2 (or PHP 8.1) via Package Center
1) Open Web Station - navigate to Script Language Settings and click create
![image](https://github.com/kimai/kimai/assets/51003649/bba773a6-da1e-413b-acc2-b247dc96d143)

2) give your profile a name and description and choose the installed PHP
![image](https://github.com/kimai/kimai/assets/51003649/89e226dc-cc98-402a-adc4-268f1422b4e7)

3) select the required extensions (I just activated all)
![image](https://github.com/kimai/kimai/assets/51003649/b4531aaf-7211-443c-9831-ddaaef4e9b10)

4) the next two windows (FPM settings and core settings) can remain unchanged - just click next

5) In the final window click `create`
![image](https://github.com/kimai/kimai/assets/51003649/1541071d-1f2f-43dc-97c9-48a971342ae3)

6) switch to Web Services and click create
![image](https://github.com/kimai/kimai/assets/51003649/33117136-667c-4aeb-a118-d99678927e82)

7) Select as service type `Native script language` and select PHP and the Web service you just have created
![image](https://github.com/kimai/kimai/assets/51003649/c320cef8-3c23-4e6a-bcad-ce5fff2e6de8)

8) give it a name and description (I just used the same name and description like before), choose the Kimai public folder and select Nginx as HTTP back-end server
![image](https://github.com/kimai/kimai/assets/51003649/f404eeb0-253c-4ea3-8616-899add79f185)

9) click create on the final window
![image](https://github.com/kimai/kimai/assets/51003649/062d8bd5-354a-4141-b13d-bb76de7a12ac)

10) and confirm reading rights of the http group on the information box shown
![image](https://github.com/kimai/kimai/assets/51003649/c921389d-f5c9-46a6-b977-d59e214eb393)

11) switch to web portal and click create
![image](https://github.com/kimai/kimai/assets/51003649/53639070-966b-4469-abf9-889e98ea440c)

11) chose Web service portal as your portal type
![image](https://github.com/kimai/kimai/assets/51003649/67d2cead-ed42-4659-b5ca-5f66838dd885)

12) Select as Service the service you just have created and choose your portal type (name-based or port-based)
![image](https://github.com/kimai/kimai/assets/51003649/4557e2c6-74cc-47c8-bda3-61ed7e9acfd2)

13) Now, your settings in Web Station are finished and the hardest part is next to come
![image](https://github.com/kimai/kimai/assets/51003649/6dae45f5-50f1-48f2-ada5-ce39facabe47)

14) SSH into your NAS and navigate to `/etc/nginx/conf.d`
There, you will find a file like `.service.6013e6a4-c593-4a01-a0ad-df30eb921aa3.e0faed91-374e-444f-a2fc-283c6ddf4e1b.conf`
When you open it, it starts with the kimai root folder you have set, in the example above with  `root    "/volume1/web/kimai/public";`  (if not, you have to open each `.service...conf` file until you find it)
![image](https://github.com/kimai/kimai/assets/51003649/0e6a2577-bcfd-4cbe-bcc4-62c63efa1f0c)

Make sure to have the right file and scroll down to the bottom of the file; it ends with an include like
`include /usr/local/etc/nginx/conf.d/e0faed91-374e-444f-a2fc-283c6ddf4e1b/user.conf*;`
![image](https://github.com/kimai/kimai/assets/51003649/e0c8ec91-4e22-44a2-aad2-b165a2f62085)

15) create the folder under `/usr/local/etc/nginx/conf.d` so that you the folder referred to in the conf file above, in my example the folder `/usr/local/etc/nginx/conf.d/e0faed91-374e-444f-a2fc-283c6ddf4e1b` 

16) place your nginx-user-conf in this folder - I named my `user.conf.kimai` and it has the following content:
```
access_log off;
log_not_found off;

location ~ /\.ht {
    deny all;
}

location / {
    try_files $uri /index.php$is_args$args;
}

location ~ \.php$ {
    return 404;
}
```
17) restart nginx with `sudo nginx -s reload `and it hopefully works
